### ISSUE TITLE
TLDs missing in imports

### ISSUE THREAD

**Issue by Thrameos** [jpype-project/jpype#512]
Currently we register ``com``, ``gov``, ``java``, and ``org`` as tlds for importing.  However, this leaves off ``edu``, ``net``, ``mil`` and ``int`` from the original tlds.  

Per the java naming conventions

> The first component of a unique package name is always written in all-lowercase ASCII letters and should be one of the top level domain names, currently com, edu, gov, mil, net, org, or one of the English two-letter codes identifying countries as specified in ISO Standard 3166, 1981. For more information, refer to the documents stored at ftp://rs.internic.net/rfc, for example, rfc920.txt and rfc1032.txt.

Thus it seems like we should add ``edu``, ``mil``, and ``net`` to the list of expected tlds.  Anyone have an objection?  I would hope the because of Jython, those should all be available to use.

**Comment by marscher** [comment]
Sorry I've overlooked this issue completely. I always thought that this naming pattern is just a convention and so it does not need to be obeyed by libraries in general. Why would we want the US military domain being registered by default? It looks weird to me. What are the real implications of not having these?

**Comment by Thrameos** [comment]
Well it would save me from typing ``registerDomain("mil")`` every time that I am working on a military code testing (not actually kidding).   The direct implication is that Java defined 6 tlds for packages to be defined in.  And Python adopted those as reserved in Jython so we are not likely to ever run into conflicts.  (Though I would be hard pressed to find the email chain that it was discussed without trolling through the archives so lets assume I am mistaken.)  Thus lets break these down.

I know that the missing "edu" one causes issues for others who expect it to be there as it has appeared in some email and forum posts.  It is more common than "gov" on maven searches.  Thus I don't think anyone can object to it.  

"mil" is oddly specific and outside of US government contractors noone is likely to every encounter it.   A search of maven [repos](https://mvnrepository.com/search?q=mil.) finds a few mil.navy and mil.army packages so it does happen.  Not having it is probably fine.

"net" is the one I had the greatest issue.  It would look like the most likely place to have a package conflict in the future.  But I am guessing the conflict with Jython is likely to prevent it from being adopted by CPython.  It is by far the most common on [maven](https://mvnrepository.com/search?q=net.)  through many packages drop the net prefix.
 
That said not many people actually use the Java naming conventions partially because tools like maven seem to actively discourage their use.  At least in my shop everything actually has a tld such as "gov", "edu", "com" and even "mil".

**Comment by Thrameos** [comment]
One other important piece is that the naming of Java packages is dictated in the Java language specification on page 137.  The language does not force the convention but they were pretty specific as the intent.

https://docs.oracle.com/javase/specs/jls/se11/jls11.pdf

“The first component of a unique package name is always written in all-lowercase ASCII letters and should be one of the top level domain names, such as com, edu, gov, mil, net, or org, or one of the English two-letter codes identifying countries as specified in ISO Standard 3166.”

Put simply the JVM has no way to disambiguate package conflicts and was designed with this in mind.   Not that I don’t appreciate JarJarLinks for rescuing us when people fail to follow through on this or in a pinch the use of “asm” package.  There is no reason (other than vanity) that I can see for deciding to name your package “snazzy” when “com.mycompany.snazzy” would save someone from conflicting with yours (though obvious both parties are guilty in that case).
